packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -762,6 +767,78 @@ class _DialPainter extends CustomPainter {
     canvas.restore();
   }
 
+  static const double _kSemanticNodeSizeScale = 1.5;
+
+  /// Creates semantics nodes for the hour/minute labels painted on the dial.
+  ///
+  /// The nodes are positioned on top of the text and their size is
+  /// [_kSemanticNodeSizeScale] bigger than those of the text boxes to provide
+  /// bigger tap area.
+  List<SemanticsNode> graftSemantics(RenderBox renderBox) {
+    final Size size = renderBox.size;
+    final double radius = size.shortestSide / 2.0;
+    final Offset center = new Offset(size.width / 2.0, size.height / 2.0);
+    const double labelPadding = 24.0;
+    final double outerLabelRadius = radius - labelPadding;
+    final double innerLabelRadius = radius - labelPadding * 2.5;
+
+    Offset getOffsetForTheta(double theta, _DialRing ring) {
+      double labelRadius;
+      switch (ring) {
+        case _DialRing.outer:
+          labelRadius = outerLabelRadius;
+          break;
+        case _DialRing.inner:
+          labelRadius = innerLabelRadius;
+          break;
+      }
+      return center + new Offset(labelRadius * math.cos(theta),
+          -labelRadius * math.sin(theta));
+    }
+
+    final List<SemanticsNode> nodes = <SemanticsNode>[];
+
+    void paintLabels(List<TextPainter> labels, _DialRing ring) {
+      if (labels == null)
+        return;
+      final double labelThetaIncrement = -_kTwoPi / labels.length;
+      double labelTheta = math.PI / 2.0;
+
+      for (TextPainter label in labels) {
+        final double width = label.width * _kSemanticNodeSizeScale;
+        final double height = label.height * _kSemanticNodeSizeScale;
+        final Offset nodeOffset = getOffsetForTheta(labelTheta, ring) + new Offset(-width / 2.0, -height / 2.0);
+        final SemanticsNode node = new SemanticsNode(showOnScreen: renderBox.showOnScreen)
+          ..rect = new Rect.fromLTRB(
+            nodeOffset.dx,
+            nodeOffset.dy,
+            nodeOffset.dx + width,
+            nodeOffset.dy + height
+          )
+          ..updateWith(
+              config: new SemanticsConfiguration()
+                ..label = label.text.text
+                ..textDirection = textDirection
+                ..addAction(SemanticsAction.tap, () {
+                  // The action is sent to the gesture detector surrounding the
+                  // custom painter.
+                })
+          )
+          ..tags = new Set<SemanticsTag>.from(const <SemanticsTag>[
+            const SemanticsTag('dial-label'),
+          ]);
+
+        nodes.add(node);
```

Comment:
I assume the nodes are traversed in geometrical order, not time order?

cc @gspencergoog who is working on making that fixable -- fixing the order here would be a good proof of that patch.

Reply:
Aye! @gspencergoog, LMK if you need help.